src/cal/mod.rs

```diff
@@ -0,0 +1,29 @@
+pub mod types;
+
+pub use self::types::*;
+use std::collections::btree_set::Range;
+use std::collections::BTreeSet;
+use std::ops::RangeBounds;
+
+pub struct Cal {
+    events: BTreeSet<types::Event>,
+}
+
+impl Cal {
+    pub fn new() -> Cal {
+        Cal {
+            events: BTreeSet::new(),
+        }
+    }
+
+    pub fn range<R>(&self, range: R) -> Range<Event>
```

Comment:
Rename to something that is more clear for a calendar. Maybe `events_in`?

Also this function doesn't need to be generic. You could just use std::ops::Range as the argument.

Reply:
note: https://doc.rust-lang.org/reference/expressions/range-expr.html